chemprop/cli/train.py

```diff
@@ -733,11 +734,18 @@ def train_model(args, train_loader, val_loader, test_loader, output_dir, scaler)
 
         if test_loader is not None:
             if args.task_type == "regression":
-                model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
-                model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
+                model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(1, -1))
+                model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(1, -1))
             results = trainer.test(model, test_loader)[0]
             logger.info(f"Test results: {results}")
 
+            if args.save_preds:
+                preds = trainer.predict(model, test_loader)
+                preds_list = [pred.flatten() for pred in np.rollaxis(torch.stack(preds).numpy(),-1)]
+                columns = get_column_names(args.data_path, args.smiles_columns, args.reaction_columns, args.target_columns, args.ignore_columns, args.no_header_row)
+                df_preds = pd.DataFrame(list(zip(test_loader.dataset.smiles, *preds_list)), columns = columns)
+                df_preds.to_csv(model_output_dir / "test_predictions.csv", index=False)
```

Comment:
```suggestion
                predss = trainer.predict(model, test_loader)
                preds = torch.concat(predss, 0).numpy()
                columns = get_column_names(args.data_path, args.smiles_columns, args.reaction_columns, args.target_columns, args.ignore_columns, args.no_header_row)
                df_preds = pd.DataFrame(list(zip(test_loader.dataset.smiles, *preds.T)), columns = columns)
                df_preds.to_csv(model_output_dir / "test_predictions.csv", index=False)
```
This makes the style consistent with `predict.py`.

Reply:
This is much cleaner and removes the need to specifically import numpy. I have committed the suggested changes.